### PR TITLE
Upgrade to React 15.2.1

### PR DESCRIPTION
**Pull request by benvinegar**
Blockers:
- [x] ~~Stream doesn't render (React internal error)~~
- [x] ~~New warnings, deprecation notices result in noisy console~~ (good enough)
- [x] Fix tests

/cc @getsentry/team

---

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry/3689)

### CODE REVIEW

**Comment by mitsuhiko** [comment]
Looks good but we need to figure out why the stream is broken :)

---

Review status: 0 of 7 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3689#-:-KMa9KX3p3_kjQbG9nG0:99441867)*

File: tests/js/spec/views/stream/searchBar.spec.jsx [hunk 1]
```diff
@@ -216,10 +216,8 @@ describe('SearchBar', function() {
       };
       let searchBar = mount(<SearchBar {...props} />).instance();
       searchBar.updateAutoCompleteItems();
-      expect(searchBar.state.searchTerm).to.eql('unresolved');
-      expect(searchBar.state.searchItems.length).to.eql(1);
-      expect(searchBar.state.searchItems[0].desc).to.eql('unresolved');
-      expect(searchBar.state.searchItems[0].value).to.eql('unresolved');
+      expect(searchBar.state.searchTerm).to.eql('fu');
+      expect(searchBar.state.searchItems.length).to.eql(0);
```
Comment (benvinegar) [inline tests/js/spec/views/stream/searchBar.spec.jsx]: The original assertions on this test were ... completely wrong, and I'm not sure what I was doing when I wrote them. The new assertions matches the behavior from the live app today.

**Comment by benvinegar** [comment]
Okay, this is ready now.

**Comment by macqueen** [comment]
Reviewed 6 of 7 files at r2, 6 of 7 files at r3, 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry/3689)*

**Pull request merged**